File: fill_membrane.py
import numpy as np

membrane_coords = np.loadtxt("5991_L2_ts001_1.5.csv", delimiter=",")
coeffs_1 = np.loadtxt("coeffs_1.csv", delimiter=",")
coeffs_2 = np.loadtxt("coeffs_2.csv", delimiter=",")

x = membrane_coords[:,0]
y = membrane_coords[:,1]
z = membrane_coords[:,2]

x_min = np.min(x)
y_min = np.min(y)
z_min = np.min(z)
x_max = np.max(x)
y_max = np.max(y)
z_max = np.max(z)

print(membrane_coords.shape)
print(coeffs_1.shape)
print(coeffs_2.shape)

x_range = np.arange(x_min,x_max+1)
print(x_range.shape)
y_range = np.arange(y_min,y_max+1)
print(y_range.shape)
z_range = np.arange(z_min,z_max+1)

x_grid = x_range[:,None] + np.zeros((y_range.shape[0]))
print(x_grid.shape)

y_grid = y_range[:,None] + np.zeros((x_range.shape[0]))
print(y_grid.shape)
y_grid = y_grid.T

np.stack((x_grid[:,:,None], y_grid[:,:,None], axis=2)

File: fill_membrane.m
coeffs_1 = coeffvalues(surface1)
coeffs_2 = coeffvalues(surface2)

p00_1 = coeffs_1(1)
p10_1 = coeffs_1(2)
p01_1 = coeffs_1(3)
p20_1 = coeffs_1(4)
p11_1 = coeffs_1(5)
p30_1 = coeffs_1(6)
p21_1 = coeffs_1(7)
p40_1 = coeffs_1(8)
p31_1 = coeffs_1(9)

p00_2 = coeffs_2(1)
p10_2 = coeffs_2(2)
p01_2 = coeffs_2(3)
p20_2 = coeffs_2(4)
p11_2 = coeffs_2(5)
p30_2 = coeffs_2(6)
p21_2 = coeffs_2(7)
p40_2 = coeffs_2(8)
p31_2 = coeffs_2(9)


csvwrite("coeffs_1.csv", coeffs_1)
csvwrite("coeffs_2.csv", coeffs_2)

% fill = double.empty(0,3)
% 
% for i = x_min : 5 : x_max
%     for j = y_min : 5 : y_max
%         for k = z_min  : 5 : z_max
%             k1 = p00_1 + p10_1 * i + p01_1 * j + p20_1 * i^2 + p11_1 * i * j
%             k2 = p00_2 + p10_2 * i + p01_2 * j + p20_2 * i^2 + p11_2 * i * j
%             bigK = max([k1,k2])
%             smallK = min([k1,k2])
% 
%             if k > smallK && k < bigK
%                 fill = vertcat(fill, [i,j,k])
%             end
%         end
%     end
% end